File: leetcode/0120-minimumTotal.java
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        int[] dp = new int[triangle.size()+1];
        for (int i = dp.length-2; i >= 0; i--) {
            List<Integer> row = triangle.get(i);
            for (int j = 0; j <= i; j++) {
                dp[j] = Math.min(dp[j], dp[j+1]) + row.get(j);
            }
        }
        return dp[0];
    }
}

File: leetcode/0120-minimumTotal.py
class Solution:
    def minimumTotal(self, triangle):
        """
        :type triangle: List[List[int]]
        :rtype: int
        """
        res = [0 for _ in range(len(triangle)+1)]
        for i in range(len(triangle)-1, -1, -1):
            row = triangle[i]
            for j in range(len(row)):
                res[j] = min(res[j], res[j+1]) + row[j]
        return res[0]
        
